nio_cli/cli.py

```diff
@@ -2,39 +2,43 @@
 nio
 
 Usage:
-  nio new <project-name> [(--template <template> | -t <template>)] [(--pubkeeper-hostname=HOSTAME --pubkeeper-token=TOKEN)] [--username=] [--password=]
+  nio [options] new <project-name> [(--template <template> | -t <template>)]
   nio newblock <block-name>
-  nio [options] add <block-repo>... [(--upgrade | -u)]
-  nio [options] (list | ls) services [--username=] [--password=]
-  nio [options] (list | ls) blocks [--username=] [--password=]
-  nio [options] (command | co) <command-name> [<service-name>] [<block-name>] [--args=<args>] [--username=] [--password=]
-  nio [options] (config | cfg) [project] [--username=] [--password=]
-  nio [options] (config | cfg) services <service-name> [--username=] [--password=]
-  nio [options] (config | cfg) blocks <block-name> [--username=] [--password=]
-  nio [options] clone services <service-name> <new-name> [--username=] [--password=]
+  nio add <block-repo>... [(--upgrade | -u)]
+  nio [options] (list | ls) services
+  nio [options] (list | ls) blocks
+  nio [options] (command | co) <command-name> [--args=<args> (<service-name> | <block-name>)]
+  nio [options] (config | cfg)
+  nio [options] (config | cfg) services <service-name>
+  nio [options] (config | cfg) blocks <block-name>
+  nio [options] clone services <service-name> <new-name>
   nio [options] buildspec <repo-name>
   nio [options] buildreadme
   nio [options] buildrelease <repo-name>
   nio [options] blockcheck
-  nio [options] shutdown [--username=] [--password=]
-  nio [options] add_user [<project-name>] [--username=] [--password=]
-  nio [options] remove_user [<project-name>] [--username=]
+  nio [options] shutdown
+  nio [options] add_user <username> <password>
+  nio [options] remove_user <username>
   nio -h | --help
   nio --version
 
 Options:
-  -p PORT --port=PORT               Specify n.io port [default: 8181].
-  -i IP --ip=IP                     Specify n.io ip address [default: 127.0.0.1].
-  --username=Admin                  Specify username [default: Admin].
-  --password=Admin                  Specify password [default: Admin].
+  -p PORT --port=PORT               Specify nio port [default: 8181].
+  -i IP --ip=IP                     Specify nio ip address [default: 127.0.0.1].
+  --username=USERNAME               Specify username [default: Admin].
+  --password=PASSWORD               Specify password [default: Admin].
+  --project=PROJECT                 Specify project directory [default: .].
+  --pubkeeper-hostname=HOST         Specify Pubkeeper Server Hostname.
+  --pubkeeper-token=TOKEN           Specify Pukeeper Server Authentication Token.
```

Comment:
For `[options]` to work, these can only be declared in this Options section and cannot be used in the Usage section

Reply:
The `[options]` section should be for global options, not every single option (if we can avoid it). Are you saying we have to put command-specific options in here too? Like `--pubkeeper-token`? That's surprising and unfortunate. I'm not reproducing that behavior on the [docopt try site](http://try.docopt.org/) though. It seems to me this should work:
```
Usage:
  nio [options] new <project-name> [(--template <template> | -t <template>)] [--pubkeeper-hostname=HOST] [--pubkeeper-token=TOKEN]
```

Reply:
[This](http://try.docopt.org/?doc=Naval+Fate.%0D%0A%0D%0AUsage%3A%0D%0A++naval_fate.py+ship+new+%3Cname%3E...%0D%0A++naval_fate.py+ship+%3Cname%3E+move+%3Cx%3E+%3Cy%3E+%5B--speed%3D%3Ckn%3E%5D%0D%0A++naval_fate.py+ship+shoot+%3Cx%3E+%3Cy%3E%0D%0A++naval_fate.py+mine+%28set%7Cremove%29+%3Cx%3E+%3Cy%3E+%5B--moored%7C--drifting%5D%0D%0A++naval_fate.py+-h+%7C+--help%0D%0A++naval_fate.py+--version%0D%0A++naval_fate.py+%5Boptions%5D+test%0D%0A%0D%0AOptions%3A%0D%0A++-h+--help+++++Show+this+screen.%0D%0A++--version+++++Show+version.%0D%0A++--speed%3D%3Ckn%3E++Speed+in+knots+%5Bdefault%3A+10%5D.%0D%0A++--moored++++++Moored+%28anchored%29+mine.%0D%0A++--drifting++++Drifting+mine.%0D%0A%0D%0A&argv=ship+test+--speed%3D20) is kinda the weird example I was running into. If an option is attached to one of the usage statements. It cannot be pulled from `[options]` for other commands. `--pubkeeper-hostname/token` commands can be used in both `new` and `config` so I thought it would be better to keep those in the options.

Reply:
Oh I see what you mean. Ugh yeah that is lame. In this case I would actually add both `--pubkeeper-hostname` and `--pubkeeper-token` to both the `new` and `config` commands then. Even if they exist on both.

Reply:
Gotcha. I'll do that!